### PR TITLE
Support generic oauth ca cert.

### PR DESCRIPTION
**Pull request by youngm**
PR for issue: https://github.com/concourse/concourse/issues/1914

Adds support for setting the generic oauth ca cert via bosh property.

### CODE REVIEW

File: jobs/atc/templates/generic_oauth_ca_cert.erb [hunk 1]
```diff
@@ -0,0 +1 @@
+<%= p("generic_oauth.ca_cert", "") %>
```
Comment (cppforlife) [inline jobs/atc/templates/generic_oauth_ca_cert.erb]: dont need "" here if default is set in spec.

**Comment by youngm** [comment]
Thanks @cppforlife I removed `""` from the spec to match most of the other CA config.

File: jobs/atc/templates/generic_oauth_ca_cert.erb (hunk 1 above)
Comment (youngm) [reply jobs/atc/templates/generic_oauth_ca_cert.erb]: Thanks.  Removed "" from spec.

File: jobs/atc/templates/atc_ctl.erb [hunk 2]
```diff
@@ -185,6 +186,7 @@ case $1 in
       <% end %> \
       --generic-oauth-token-url <%= esc(p("generic_oauth.token_url")) %> \
       --generic-oauth-display-name <%= esc(p("generic_oauth.display_name")) %> \
+      --generic-oauth-ca-cert $GENERIC_OAUTH_CA_CERT \
```
Comment (vito) [inline jobs/atc/templates/atc_ctl.erb]: this should be passed conditionally since if it's not set it'll be pointing to an empty file, which isn't valid

**Comment by youngm** [comment]
Thanks for catching that @vito  I made the requested change.

**Comment by vito** [comment]
thanks!

**Pull request merged**